packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -0,0 +1,645 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { escape, pull } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	KeyboardShortcuts,
+	PanelBody,
+	Popover,
+	TextControl,
+	TextareaControl,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
+import { rawShortcut, displayShortcut } from '@wordpress/keycodes';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	BlockControls,
+	InnerBlocks,
+	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
+	InspectorControls,
+	RichText,
+	__experimentalLinkControl as LinkControl,
+	useBlockProps,
+	store as blockEditorStore,
+	getColorClassName,
+} from '@wordpress/block-editor';
+import { isURL, prependHTTP, safeDecodeURI } from '@wordpress/url';
+import {
+	Fragment,
+	useState,
+	useEffect,
+	useRef,
+	createInterpolateElement,
+} from '@wordpress/element';
+import { placeCaretAtHorizontalEdge } from '@wordpress/dom';
+import { link as linkIcon } from '@wordpress/icons';
+import { store as coreStore } from '@wordpress/core-data';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { ItemSubmenuIcon } from './icons';
+import { name } from './block.json';
+
+const ALLOWED_BLOCKS = [ 'core/navigation-link', 'core/navigation-submenu' ];
+
+const MAX_NESTING = 5;
```

Comment:
Should this be filtered somehow?

Reply:
I've set it to be the same as the current submenus; we can revisit it later if needed. I don't _think_ that's been discussed anywhere, so if you feel strongly about it by all means go ahead and open an issue 😄